ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/Multipart.kt

```diff
@@ -196,9 +196,12 @@ suspend fun boundary(boundaryPrefixed: ByteBuffer, input: ByteReadChannel): Bool
 
 /**
  * Skip multipart boundary
+ * @return `true` if end channel encountered
  */
 private suspend fun skipBoundary(boundaryPrefixed: ByteBuffer, input: ByteReadChannel): Boolean {
-    input.skipDelimiter(boundaryPrefixed)
+    if (!input.trySkipDelimiter(boundaryPrefixed)) {
```

Comment:
changed behavior without test

Reply:
Actually two tests were added

Reply:
and no one is testing this method directly

Reply:
Added tests